scipy/signal/_spectral_py.py

```diff
@@ -1891,27 +1908,33 @@ def detrend_func(d):
         raise ValueError('Unknown scaling: %r' % scaling)
 
     if mode == 'stft':
-        scale = np.sqrt(scale)
+        scale = xp.sqrt(scale)
 
     if return_onesided:
-        if np.iscomplexobj(x):
+        try:
+            is_complex = xp.iscomplexobj(x)
```

Comment:
Looks like we should add an `is_complex` to `_array_api.py`, which I think would be `x.dtype in {xp.complex64, xp.complex128}`.